x/auction/keeper/msg_server_reveal_bid.go

```diff
@@ -0,0 +1,77 @@
+package keeper
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/crypto"
+
+	"github.com/Stride-Labs/stride/v5/x/auction/types"
+)
+
+// Bidders send in the bid object with the original salt used to encrypt their sealed bid from earlier messages
+// If their sealed bid hash matches the hash we generate then nothing has changed and their revealed bid is entered
+
+func (k msgServer) RevealBid(goCtx context.Context, msg *types.MsgRevealBid) (*types.MsgRevealBidResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	ctx.Logger().Info(fmt.Sprintf("[auction]"))
+	ctx.Logger().Info(fmt.Sprintf("[auction] Reveal request for %+v", msg))
+
+	auctionPools := k.Keeper.GetAllAuctionPool(ctx)
+	for _, pool := range auctionPools {
```

Comment:
We should remove this for loop and use GetAuctionPool function instead to get auction pool by id.